pkg/identity/issuerpool.go

```diff
@@ -41,11 +41,11 @@ func (p IssuerPool) Authenticate(ctx context.Context, token string, opts ...conf
 }
 
 func extractIssuerURL(token string) (string, error) {
-	parts := strings.Split(token, ".")
-	if len(parts) != 3 {
-		return "", fmt.Errorf("oidc: malformed jwt, expected 3 parts got %d", len(parts))
+	if strings.Count(token, ".") != 2 {
```

Comment:
You might want to be a bit more robust and handle not only regular JWTs but also SD JWTs, which have recently been graduated to a proposed standard ([RFC 9901](https://datatracker.ietf.org/doc/rfc9901/)).

An SD JWT has two "." or four "." in those cases when it includes a SD-JWT+KB proof (a cryptographic key binding proof).